src/Header/Cookie.php

```diff
@@ -97,13 +97,30 @@ public function getFieldValue()
     {
         $nvPairs = [];
 
-        foreach ($this as $name => $value) {
+        $cookies = [];
+        foreach ($this->flattenCookies($cookies, $this) as $name => $value) {
             $nvPairs[] = $name . '=' . (($this->encodeValue) ? urlencode($value) : $value);
         }
 
         return implode('; ', $nvPairs);
     }
 
+    protected function flattenCookies(&$result, $data, $prefix = null)
```

Comment:
Btw, we discourage the usage of passing value by reference in function. A good use case for reference is performance reason, but this is not the case.